Mathlib/Data/Matroid/IndepAxioms.lean

```diff
@@ -0,0 +1,565 @@
+but can be described just as naturally via their collections of independent sets,
+and in fact such a description, being more 'verbose', can often be useful.
+As well as this, the definition of a `Matroid` uses an unwieldy 'maximality'
+axiom that can be dropped in cases where there is some finiteness assumption.
+
+This file provides several ways to do define a matroid in terms of its independence or base
+predicates, using axiom sets that are appropriate in different settings,
+and often much simpler than the general definition.
+It also contains `simp` lemmas and typeclasses as appropriate.
+
+All the independence axiom sets need nontriviality (the empty set is independent),
+monotonicity (subsets of independent sets are independent),
+and some form of 'augmentation' axiom, which allows one to enlarge a non-maximal independent set.
+This augmentation axiom is still required when there are finiteness assumptions, but is simpler.
+It just states that if `I` is a finite independent set and `J` is a larger finite
+independent set, then there exists `e ∈ J \ I` for which `insert e I` is independent.
+This is the axiom that appears in all most of the definitions.
+
+## Main definitions
+
+* `Matroid.matroidOfBase` constructs a matroid from the base axioms (using the definition of a
+  matroid).
+
+* `Matroid.matroidOfIndep` constructs a matroid described in terms of its independent sets
+  in full generality, using infinite versions of the axioms.
+
+* `Matroid.matroidOfIndepOfFinitary` constructs a `Finitary` matroid in terms of its independent
+  sets in the special case where independence of a set is determined only by that of its
+  finite subsets. This construction uses Zorn's lemma.
+
+* `Matroid.matroidOfIndepOfBdd` constructs a matroid in terms of its independence sets in
+  the case where there is some known absolute upper bound on the size of an independent set.
+  This uses the infinite version of the augmentation axiom, and gives a `FiniteRk` matroid.
+
+* `Matroid.matroidOfIndepOfBddAugment` is the same as the above, but with a finite augmentation
+  axiom.
+
+* `Matroid.matroidOfExistsFiniteBase` constructs a matroid from its bases, if it is known that one
+  of them is finite. This gives a `FiniteRk` matroid.
+
+* `Matroid.matroidOfBaseOfFinite` constructs a `Finite` matroid from its bases.
+
+* `Matroid.matroidOfIndepOfFinite` constructs a `Finite` matroid from its independent sets.
+
+* `Matroid.matroidOfIndepFinset` constructs a `Finitary` matroid on `α` from an independence
+  predicate on `Finset α`.
+
+* `Matroid.matroidOfIndepOfExistsMatroid` constructs a 'copy' of a matroid that is known only
+  existentially, but whose independence predicate is known explicitly.
+-/
+
+open Set
+
+variable {α : Type _}
+
+namespace Matroid
+
+/-- A constructor for matroids via the base axioms.
+  (In fact, just a wrapper for the definition of a matroid) -/
+def matroidOfBase (E : Set α) (Base : Set α → Prop) (exists_base : ∃ B, Base B)
```

Comment:
```suggestion
protected def ofBase (E : Set α) (Base : Set α → Prop) (exists_base : ∃ B, Base B)
```
so that the caller can write either `Matroid.ofBase` or just `.ofBase`, and not have to write `Matroid` twice,

Reply:
Good idea - thanks!